Modules/TPC/src/ClusterVisualizer.cxx

```diff
@@ -0,0 +1,181 @@
+
+namespace o2::quality_control_modules::tpc
+{
+
+void ClusterVisualizer::configure(std::string name, const boost::property_tree::ptree& config)
+{
+
+  for (const auto& timestamp : config.get_child("qc.postprocessing." + name + ".timestamps")) {
+    mTimestamps.emplace_back(std::stol(timestamp.second.data()));
+  }
+
+  std::vector<std::string> keyVec{};
+  std::vector<std::string> valueVec{};
+  for (const auto& data : config.get_child("qc.postprocessing." + name + ".lookupMetaData")) {
+    mLookupMaps.emplace_back(std::map<std::string, std::string>());
+    if (const auto& keys = data.second.get_child_optional("keys"); keys.has_value()) {
+      for (const auto& key : keys.value()) {
+        keyVec.emplace_back(key.second.data());
+      }
+    }
+    if (const auto& values = data.second.get_child_optional("values"); values.has_value()) {
+      for (const auto& value : values.value()) {
+        valueVec.emplace_back(value.second.data());
+      }
+    }
+    auto vecIter = 0;
+    if ((keyVec.size() > 0) && (keyVec.size() == valueVec.size())) {
+      for (const auto& key : keyVec) {
+        mLookupMaps.back().insert(std::pair<std::string, std::string>(key, valueVec.at(vecIter)));
+        vecIter++;
+      }
+    }
+    if (keyVec.size() != valueVec.size()) {
+      throw std::runtime_error("Number of keys and values for lookupMetaData are not matching");
+    }
+    keyVec.clear();
+    valueVec.clear();
+  }
+
+  for (const auto& data : config.get_child("qc.postprocessing." + name + ".storeMetaData")) {
+    mStoreMaps.emplace_back(std::map<std::string, std::string>());
+    if (const auto& keys = data.second.get_child_optional("keys"); keys.has_value()) {
+      for (const auto& key : keys.value()) {
+        keyVec.emplace_back(key.second.data());
+      }
+    }
+    if (const auto& values = data.second.get_child_optional("values"); values.has_value()) {
+      for (const auto& value : values.value()) {
+        valueVec.emplace_back(value.second.data());
+      }
+    }
+    auto vecIter = 0;
+    if ((keyVec.size() > 0) && (keyVec.size() == valueVec.size())) {
+      for (const auto& key : keyVec) {
+        mStoreMaps.back().insert(std::pair<std::string, std::string>(key, valueVec.at(vecIter)));
+        vecIter++;
+      }
+    }
+    if (keyVec.size() != valueVec.size()) {
+      throw std::runtime_error("Number of keys and values for storeMetaData are not matching");
+    }
+    keyVec.clear();
+    valueVec.clear();
+  }
+
+  for (const auto& entry : config.get_child("qc.postprocessing." + name + ".histogramRanges")) {
+    for (const auto& type : entry.second) {
+      for (const auto& value : type.second) {
+        mRanges[type.first].emplace_back(std::stof(value.second.data()));
+      }
+    }
+  }
+
+  mPath = config.get<std::string>("qc.postprocessing." + name + ".path");
+}
+
+void ClusterVisualizer::initialize(Trigger, framework::ServiceRegistry&)
+{
+  mCdbApi.init("ccdb-test.cern.ch:8080");
```

Comment:
Shouldn't this be configurable from the json?

Reply:
absolutely